lib/syntax.ml

```diff
@@ -19,12 +19,15 @@ type env = {
 type res_expr =
   | ResValue of Type.t (** literal value *)
   | ResParam of param
+  | ResInTupleList of param_id * res_in_tuple_list
   | ResInparam of param
   | ResChoices of param_id * res_expr choices
   | ResInChoice of param_id * [`In | `NotIn] * res_expr
   | ResFun of Type.func * res_expr list (** function kind (return type and flavor), arguments *)
   [@@deriving show]
-
+  
```

Comment:
I could not add a clone just by calling `assign_types` from `resolve_columns`, but I decided not to do it, bypassing
the following function:

```ocaml
and resolve_types env expr =
  let expr = resolve_columns env expr in
  try
    assign_types env expr
  with
    exn ->
      eprintfn "resolve_types failed with %s at:" (Printexc.to_string exn);
      eprintfn "%s" (show_res_expr expr);
      raise exn
```